dpdk/dpdk_vrouter.c

```diff
@@ -155,7 +153,20 @@ dpdk_core_mask_get(void) {
             cpu_core_mask |= (uint64_t)1 << i;
     }
 
-    if(!cpu_core_mask)
+    if (!cpu_core_mask)
+        return VR_DPDK_DEF_LCORE_MASK;
+
+    /*
+     * Do not allow to run vRouter on all the cores available, as some have
+     * to be left for virtual machines.
+     */
+    system_cpus_count = sysconf(_SC_NPROCESSORS_CONF);
+    if (system_cpus_count == -1)
+        return VR_DPDK_DEF_LCORE_MASK;
+
+    core_mask_count
+        = __builtin_popcountll((unsigned long long)cpu_core_mask);
+    if (core_mask_count == system_cpus_count)
```

Comment:
Might be better to fail the provisioning instead of silently reverting to default core mask.